contracts/src/challengeV2/libraries/EdgeChallengeManagerLib.sol

```diff
@@ -501,13 +509,22 @@ library EdgeChallengeManagerLib {
             if (firstRivalCreatedAtBlock > edgeCreatedAtBlock) {
                 // if this edge was created before the first rival then we return the difference
                 // in createdAtBlock number
-                return firstRivalCreatedAtBlock - edgeCreatedAtBlock;
+                unrivaledTime = firstRivalCreatedAtBlock - edgeCreatedAtBlock;
             } else {
                 // if this was created at the same time as, or after the the first rival
                 // then we return 0
-                return 0;
+                unrivaledTime = 0;
             }
         }
+
+        // inherit timer here
+        if (store.edges[edgeId].lowerChildId != bytes32(0)) {
+            uint64 lowerTimer = timeUnrivaled(store, store.edges[edgeId].lowerChildId, numBigStepLevel);
+            uint64 upperTimer = timeUnrivaled(store, store.edges[edgeId].upperChildId, numBigStepLevel);
```

Comment:
this recursive loop might have problem with solidity stack limit or simply gas
we may allow optional caching so that time inheritance can be done in multiple steps

Reply:
https://github.com/OffchainLabs/bold/pull/562/commits/d80f7b7d0042063a640368d73a429fc98445f55b

Allowing caching at every edge have the drawback where an attacker can grief by caching at various spot when the timer is small, since the contract does not know the timer is stale, it would use the cached value and stop the traversal. This forces the honest validator to explicitly update those stale timer before calling confirmByTimer.

Reply:
no more recursion